File: Produtos.py

class produto:
    def __init__(self, nome, tipo, quantidade):
        self.__nome = nome
        self.__tipo = tipo
        # garante não-negativo na criação
        self.__quantidade = max(0, int(quantidade or 0))

    # getters
    def getnome(self): return self.__nome
    def gettipo(self): return self.__tipo
    def getquantidade(self): return self.__quantidade

    # Quantidades (sempre não-negativas)
    def aumentarquantidade(self, valor):
        v = max(0, int(valor or 0))   # ignora números negativos
        self.__quantidade += v

    def diminuirquantidade(self, valor):
        v = max(0, int(valor or 0))   # ignora números negativos
        # nunca deixa ir abaixo de zero
        self.__quantidade = max(0, self.__quantidade - v)

    def exibirDetalhes(self):
        return print(f"Nome: {self.__nome}\n "
                     f"Tipo: {self.__tipo}\n "
                     f"Quantidade:{self.__quantidade}\n ")

    def zerar(self):
        self.__quantidade = 0

File: Kotlin/Produtos.kt
package com.exemplo.natabase

class Produto(
    private val nome: String,
    private val tipo: String,
    quantidade: Int
) {
    private var quantidade: Int = if (quantidade >= 0) quantidade else 0

    fun getNome(): String = nome
    fun getTipo(): String = tipo
    fun getQuantidade(): Int = quantidade

    fun aumentarQuantidade(valor: Int) {
        val v = if (valor > 0) valor else 0
        quantidade += v
    }

    fun diminuirQuantidade(valor: Int) {
        val v = if (valor > 0) valor else 0
        quantidade = (quantidade - v).coerceAtLeast(0)
    }

    fun exibirDetalhes(): String {
        return "Nome: $nome\nTipo: $tipo\nQuantidade: $quantidade\n"
    }

    fun zerar() {
        quantidade = 0
    }
}
